src/target/riscv/riscv-013.c

```diff
@@ -1503,15 +1503,20 @@ static void deinit_target(struct target *target)
 	info->version_specific = NULL;
 }
 
-static int set_haltgroup(struct target *target, bool *supported)
+typedef enum {
+	HALTGROUP,
+	RESUMEGROUP
+} grouptype_t;
+static int set_group(struct target *target, bool *supported, unsigned group, grouptype_t grouptype)
 {
-	uint32_t write = set_field(DM_DMCS2_HGWRITE, DM_DMCS2_GROUP, target->smp);
+	uint32_t write = set_field(DM_DMCS2_HGWRITE, DM_DMCS2_GROUP, group);
+	write = set_field(write, DM_DMCS2_GROUPTYPE, 1);
 	if (dmi_write(target, DM_DMCS2, write) != ERROR_OK)
 		return ERROR_FAIL;
 	uint32_t read;
```

Comment:
Just a cosmetic thing: I'd recommend to use variable different names than `read` and `write`, which are already used for system calls (`man 2 read`, `man 2 write`). 

E.g. `read_val`, `write_val`.

Reply:
OK